lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -701,14 +695,9 @@ public function innerJoin(string $fromAlias, string $join, string $alias, ?strin
      */
     public function leftJoin(string $fromAlias, string $join, string $alias, ?string $condition = null)
     {
-        return $this->add('join', [
-            $fromAlias => [
-                'joinType'      => 'left',
-                'joinTable'     => $join,
-                'joinAlias'     => $alias,
-                'joinCondition' => $condition,
-            ],
-        ], true);
+        $join = Join::left($join, $alias, $condition);
+
+        return $this->add('join', [$fromAlias => $join], true);
```

Comment:
Would it make sense to keep the entire body as one expression as it was done before? This way it looks more functional/streamlined/sexy.

Reply:
Done in a894328e5eb57309c485b3d374867fbbf9020703.